<h2 style='color:green' align="center"> Predicting the profits of 50 startups based on different variables </h2>


**Dataset** - (https://www.kaggle.com/farhanmd29/50-startups)

**About this file**

This dataset has data collected from New York, California and Florida about 50 business Startups "17 in each state". The variables used in the dataset are Profit, R&D spending, Administration Spending, and Marketing Spending. 

In [1]:
import pandas as pd
df = pd.read_csv('50_startups.csv')
df.head()

R&D Spend  Administration  Marketing Spend       State     Profit
0  165349.20       136897.80        471784.10    New York  192261.83
1  162597.70       151377.59        443898.53  California  191792.06
2  153441.51       101145.55        407934.54     Florida  191050.39
3  144372.41       118671.85        383199.62    New York  182901.99
4  142107.34        91391.77        366168.42     Florida  166187.94

<h4 style='color:green'>Data Exploration </h4>


In [2]:
df.describe()

R&D Spend  Administration  Marketing Spend         Profit
count      50.000000       50.000000        50.000000      50.000000
mean    73721.615600   121344.639600    211025.097800  112012.639200
std     45902.256482    28017.802755    122290.310726   40306.180338
min         0.000000    51283.140000         0.000000   14681.400000
25%     39936.370000   103730.875000    129300.132500   90138.902500
50%     73051.080000   122699.795000    212716.240000  107978.190000
75%    101602.800000   144842.180000    299469.085000  139765.977500
max    165349.200000   182645.560000    471784.100000  192261.830000

In [3]:
# looking for any NA value
df.isnull().sum()

R&D Spend          0
Administration     0
Marketing Spend    0
State              0
Profit             0
dtype: int64

In [4]:
# Converting the state column into numerical

dummies = pd.get_dummies(df.State)
dummies.head()

California  Florida  New York
0           0        0         1
1           1        0         0
2           0        1         0
3           0        0         1
4           0        1         0

In [5]:
merged = pd.concat([df, dummies], axis = 'columns')
merged.head()

R&D Spend  Administration  Marketing Spend       State     Profit  \
0  165349.20       136897.80        471784.10    New York  192261.83   
1  162597.70       151377.59        443898.53  California  191792.06   
2  153441.51       101145.55        407934.54     Florida  191050.39   
3  144372.41       118671.85        383199.62    New York  182901.99   
4  142107.34        91391.77        366168.42     Florida  166187.94   

   California  Florida  New York  
0           0        0         1  
1           1        0         0  
2           0        1         0  
3           0        0         1  
4           0        1         0

In [6]:
# Dropping california because of dummy variable trap

df2 = merged.drop(['State', 'California'], axis = 'columns')
df2.head()

R&D Spend  Administration  Marketing Spend     Profit  Florida  New York
0  165349.20       136897.80        471784.10  192261.83        0         1
1  162597.70       151377.59        443898.53  191792.06        0         0
2  153441.51       101145.55        407934.54  191050.39        1         0
3  144372.41       118671.85        383199.62  182901.99        0         1
4  142107.34        91391.77        366168.42  166187.94        1         0

<h4 style='color:green'>Model building</h4>


In [7]:
x = df2.drop(['Profit'], axis = 'columns')
y = df2['Profit']

In [8]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2)

from sklearn.linear_model import LinearRegression
model = LinearRegression()
model.fit(x_train, y_train)


LinearRegression()

In [9]:
x_test

R&D Spend  Administration  Marketing Spend  Florida  New York
26   75328.87       144135.98        134050.07        1         0
30   61994.48       115641.28         91131.24        1         0
37   44069.95        51283.14        197029.42        0         0
2   153441.51       101145.55        407934.54        1         0
4   142107.34        91391.77        366168.42        1         0
48     542.05        51743.15             0.00        0         1
7   130298.13       145530.06        323876.68        1         0
39   38558.51        82982.09        174999.30        0         0
18   91749.16       114175.79        294919.57        1         0
31   61136.38       152701.92         88218.23        0         1

In [10]:
y_test

26    105733.54
30     99937.59
37     89949.14
2     191050.39
4     166187.94
48     35673.41
7     155752.60
39     81005.76
18    124266.90
31     97483.56
Name: Profit, dtype: float64

In [11]:
model.predict(x_test)

array([112127.2877645 , 102695.86879446,  93055.70668887, 183840.67409117,
       174629.34438145,  54998.64930955, 160118.70034543,  85753.23395897,
       130465.75337781,  98330.80581135])

In [12]:
model.score(x_test, y_test)

0.9662730925515729

**From above its clear that model is good to go as prediction is 96.63% correct.**